job/server/src/main/java/alluxio/master/job/plan/PlanTracker.java

```diff
@@ -96,20 +94,19 @@ public JobTracker(long capacity, long retentionMs, long maxJobPurgeCount) {
     mCoordinators = new ConcurrentHashMap<>(0,
         0.95f, ServerConfiguration.getInt(PropertyKey.MASTER_RPC_EXECUTOR_PARALLELISM));
     mFinished = new LinkedBlockingQueue<>();
-    mJobIdGenerator = new JobIdGenerator();
   }
 
-  private void statusChangeCallback(JobInfo jobInfo) {
-    if (jobInfo == null) {
+  private void statusChangeCallback(PlanInfo planInfo) {
```

Comment:
Consider passing in the new status instead of having to derive it by inspecting `PlanInfo`